packages/fhir/src/Adaptor.js

```diff
@@ -38,202 +57,219 @@ export function execute(...operations) {
+ *   name: [
  *     {
- *       fullUrl: "https://hapi.fhir.org/baseR4/Patient/592442",
- *       resource: {
- *         resourceType: "Patient",
- *         id: "592442",
- *         name: [{ given: "Caleb", family: "Cushing" }],
- *       },
- *       request: {
- *         method: "POST",
- *         url: "Patient",
- *       },
+ *       use: 'official',
+ *       family: 'La Paradisio',
+ *       given: ['Josephine', 'Nessa'],
  *     },
  *   ],
  * });
- * @function
- * @param {object} params - data to create the new transaction
+ * @param {string} resourceType - The resource type to create
+ * @param {object} resource - The resource to create
+ * @param {object} params - (Optional) FHIR parameters to control and configure resource creation
  * @param {function} callback - (Optional) callback function
  * @returns {Operation}
+ * @state {FHIRHttpState}
  */
-export function createTransactionBundle(params, callback) {
-  return state => {
-    const [resolvedParams] = expandReferences(state, params);
-
-    const { baseUrl, apiPath, authType, token } = state.configuration;
-
-    const url = `${baseUrl}/${apiPath}`;
-    const auth = `${authType} ${token}`;
+export function create(resourceType, resource, params, callback = s => s) {
+  return async state => {
+    const [resolvedResourceType, resolvedResource, resolvedParams] =
+      expandReferences(state, resourceType, resource, params);
 
-    const options = {
-      body: {
-        resourceType: 'Bundle',
-        type: 'transaction',
-        entry: resolvedParams,
-      },
-      auth,
+    const opts = {
+      ...resolvedParams,
+      body: { ...resolvedResource, resourceType: resolvedResourceType },
     };
 
-    return request(url, options, 'POST').then(({ response, data }) =>
-      handleResponse(response, data, state, callback)
+    const response = await util.request(
+      state.configuration,
+      'POST',
+      resolvedResourceType,
+      opts
     );
+    return util.prepareNextState(state, response, callback);
   };
 }
 
 /**
- * Get a resource in a FHIR system
+ * Create a transactionBundle to process multiple requests at once
  * @public
- * @example <caption>Get Claim from FHIR with optional query</caption>
- * get("Claim", { _include: "Claim:patient", _sort: "-_lastUpdated", _count: 200 })
- * @example <caption>Get Patient from FHIR</caption>
- * get('Patient');
+ * @example
+ * createTransactionBundle([
+ *   {
+ *     fullUrl: "https://hapi.fhir.org/baseR4/Patient/592442",
+ *     resource: {
+ *       resourceType: "Patient",
+ *       id: "592442",
+ *       name: [{ given: "Caleb", family: "Cushing" }],
+ *     },
+ *     request: {
+ *       method: "POST",
+ *       url: "Patient",
+ *     },
+ *   },
+ * ]);
  * @function
- * @param {string} path - Path to resource
- * @param {object} query - data to get the new resource
+ * @param {array} entries - array of transactions
+ * @param {object} params - data to create the new transaction
```

Comment:
Ack, something is wrong with the spec. My bad.

Can we remove the `params` parameter entirely? It's only adding confusion.

If we want to support extra parameters to the bundle, we should do so when we implement the other FHIR verbs (including, I now realise, `batch` and `transaction`)

Reply:
Okay got you, let me do that

Reply:
Done ✅